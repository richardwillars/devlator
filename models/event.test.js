const eventModule = require('./event');

describe('models/event', () => {
  it('should export the schema', () => {
    const mongoose = {
      model: () => {},
      Schema: class Schema {},
    };
    const eventModel = eventModule(mongoose);
    expect(eventModel.schema).toMatchSnapshot();
  });

  it('should export the model', () => {
    const modelMock = {};
    const mongoose = {
      model: jest.fn().mockReturnValue(modelMock),
      Schema: class Schema {
        constructor(props) {
          this.props = props;
        }
      },
    };
    const eventModel = eventModule(mongoose);

    expect(eventModel.model).toEqual(modelMock);
    expect(mongoose.model).toHaveBeenCalledTimes(1);
    expect(mongoose.model.mock.calls[0][0]).toEqual('Event');
    const initialisedSchema = mongoose.model.mock.calls[0][1];
    expect(initialisedSchema instanceof mongoose.Schema).toEqual(true);
    expect(initialisedSchema.props).toMatchSnapshot();
  });
});
