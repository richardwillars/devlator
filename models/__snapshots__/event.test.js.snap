// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models/event should export the model 1`] = `
Object {
  "deviceId": Object {
    "required": true,
    "type": [Function],
  },
  "driverId": Object {
    "required": true,
    "type": [Function],
  },
  "driverType": Object {
    "required": true,
    "type": [Function],
  },
  "event": Object {
    "required": true,
    "type": [Function],
  },
  "eventType": Object {
    "enum": Array [
      "request",
      "device",
    ],
    "required": true,
    "type": [Function],
  },
  "value": Object {
    "default": Object {},
    "required": false,
    "type": [Function],
  },
  "when": Object {
    "default": [Function],
    "required": false,
    "type": [Function],
  },
}
`;

exports[`models/event should export the schema 1`] = `
Object {
  "deviceId": Object {
    "required": true,
    "type": [Function],
  },
  "driverId": Object {
    "required": true,
    "type": [Function],
  },
  "driverType": Object {
    "required": true,
    "type": [Function],
  },
  "event": Object {
    "required": true,
    "type": [Function],
  },
  "eventType": Object {
    "enum": Array [
      "request",
      "device",
    ],
    "required": true,
    "type": [Function],
  },
  "value": Object {
    "default": Object {},
    "required": false,
    "type": [Function],
  },
  "when": Object {
    "default": [Function],
    "required": false,
    "type": [Function],
  },
}
`;
